src/slot_ranges.c

```diff
@@ -41,6 +41,17 @@ const SharedSlotRangeArray *Slots_GetLocalSlots(void) {
   return localSlots;
 }
 
+const SharedSlotRangeArray *Slots_Clone(const SharedSlotRangeArray *src) {
+  // remove constness - needed in order to manage the refcount
+  SharedSlotRangeArray *slots = (SharedSlotRangeArray *)src;
+  if (!slots) {
+    return NULL;
+  }
+  uint32_t refcount = atomic_fetch_add_explicit(&slots->refcount, 1, memory_order_acquire);
```

Comment:
For incrementing the ref count, can `memory_order_relaxed` suffice?

Reply:
I think we can use `relaxed` because we know the count won't reach 0 while we are here, as the caller has a valid reference.
Anyway, `acquire` is also good